From 183165fb0270e8697550f3ebdbe0699ee97e4e3a Mon Sep 17 00:00:00 2001
Message-ID: <183165fb0270e8697550f3ebdbe0699ee97e4e3a.1735936141.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1735936141.git.sharlatanus@gmail.com>
References: <cover.1735936141.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 3 Jan 2025 19:54:55 +0000
Subject: [PATCH 6/6] gnu: xtensor: Update to 0.25.0.

* gnu/packages/algebra.scm (xtensor): Update to 0.25.0.
[native-inputs]: Add nlohmann-json.

Change-Id: Iccaf397423d2a2b7b008082a9761c89ca2adc9a2
---
 gnu/packages/algebra.scm | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/gnu/packages/algebra.scm b/gnu/packages/algebra.scm
index 4d84958abf..a1217c53d3 100644
--- a/gnu/packages/algebra.scm
+++ b/gnu/packages/algebra.scm
@@ -16,6 +16,7 @@
 ;;; Copyright © 2021 Lars-Dominik Braun <ldb@leibniz-psychology.org>
 ;;; Copyright © 2022 Maxim Cournoyer <maxim.cournoyer@gmail.com>
 ;;; Copyright © 2023 Mehmet Tekman <mtekman89@gmail.com>
+;;; Copyright © 2025 Sharlatan Hellseher <sharlatanus@gmail.com>
 ;;;
 ;;; This file is part of GNU Guix.
 ;;;
@@ -1163,7 +1164,7 @@ (define-public eigen-for-tensorflow
 (define-public xtensor
   (package
     (name "xtensor")
-    (version "0.24.6")
+    (version "0.25.0")
     (source
      (origin
        (method git-fetch)
@@ -1171,7 +1172,7 @@ (define-public xtensor
              (url "https://github.com/xtensor-stack/xtensor")
              (commit version)))
        (sha256
-        (base32 "0gf5m5p61981pv7yh5425lcv8dci948ri37hn1zlli7xg54x0g3i"))
+        (base32 "0ziqybfm0fh6kr8qwxqacr04m9gm0njbn520izm6rsh9hysxsmw5"))
        (file-name (git-file-name name version))))
     (build-system cmake-build-system)
     (arguments
@@ -1181,6 +1182,7 @@ (define-public xtensor
     (native-inputs
      (list doctest
            googletest
+           nlohmann-json
            xtl))
     (home-page "https://xtensor.readthedocs.io/en/latest/")
     (synopsis "C++ tensors with broadcasting and lazy computing")
-- 
2.46.0

